htdocs/class/auth/auth_xoops.php

```diff
@@ -46,7 +46,7 @@ public function __construct(XoopsDatabase $dao = null)
      * @param  string $pwd
      * @return bool
      */
-    public function authenticate($uname, $pwd = null)
+    public function authenticate($uname = null, $pwd = null)
```

Comment:
It is upside down. The problem is in the parent.

The signature on XoopsAuth::authenticate() should include $uname at least. Something may complain about the unused parameter, but that isn't a real issue. There is no way this should be called with out $uname, that should be an error.

This is a mistake made long ago. I don't see a big problem in changing the parent signature to match reality.

This applies to all the changes to the sibling classes as well.